parser/parser.go

```diff
@@ -163,7 +173,9 @@ func Format(in []byte) ([]byte, error) {
 func FormatWithConfig(in []byte, c Config) ([]byte, error) {
 	addMetaCommentsToConfig(in, &c)
 	if c.Disable {
-		log.Infoln("Ignored file with 'disable' comment.")
+		if c.Logger != nil {
```

Comment:
Can we avoid checking for nil in every use, and instead make every public function set logger to an internal noopLogger (that doesn't do anything) if logger is nil?

Reply:
There's no good place to have such a read-write field to set to a noopLogger. The parser type isn't enough, because we need to log from some funcs which don't hang off the parser.

I assume you didn't like the repetition, so I set up two methods on the config type to deduplicate the nil checks. I don't think the nil checks harm performance, because an interface call always has to check whether the interface is nil or not.